js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js

```diff
@@ -1,36 +1,45 @@
-var noop = require("../../../core/utils/common").noop,
-    Class = require("../../../core/class"),
-    extend = require("../../../core/utils/extend").extend,
-    errors = require("../../widget/ui.errors"),
-    dateUtils = require("../../../core/utils/date"),
-    isNumeric = require("../../../core/utils/type").isNumeric,
-    typeUtils = require("../../../core/utils/type"),
-    themes = require("../../themes");
+import BasePositioningStrategy from "./ui.scheduler.appointmentsPositioning.strategy.base";
+import AdaptivePositioningStrategy from "./ui.scheduler.appointmentsPositioning.strategy.adaptive";
+
+import { noop } from "../../../core/utils/common";
+import Class from "../../../core/class";
+import { extend } from "../../../core/utils/extend";
+import errors from "../../widget/ui.errors";
+import dateUtils from "../../../core/utils/date";
+import { isNumeric } from "../../../core/utils/type";
+import typeUtils from "../../../core/utils/type";
+import themes from "../../themes";
+
 
 var toMs = dateUtils.dateToMilliseconds;
 
 var abstract = Class.abstract;
 
-var APPOINTMENT_MIN_COUNT = 1,
-    APPOINTMENT_MIN_SIZE = 2,
-
+var APPOINTMENT_MIN_SIZE = 2,
     COMPACT_APPOINTMENT_DEFAULT_WIDTH = 15,
-
     APPOINTMENT_DEFAULT_HEIGHT = 20,
-    APPOINTMENT_DEFAULT_WIDTH = 40,
 
     COMPACT_THEME_APPOINTMENT_DEFAULT_HEIGHT = 18,
-    COMPACT_THEME_APPOINTMENT_DEFAULT_OFFSET = 22,
-    COMPACT_APPOINTMENT_DEFAULT_OFFSET = 3,
-
-    DROP_DOWN_BUTTON_DEFAULT_WIDTH = 24,
 
-    DROP_DOWN_BUTTON_ADAPTIVE_SIZE = 28,
-    DROP_DOWN_BUTTON_ADAPTIVE_BOTTOM_OFFSET = 40;
+    DROP_DOWN_BUTTON_ADAPTIVE_SIZE = 28;
 
 var BaseRenderingStrategy = Class.inherit({
     ctor: function(instance) {
         this.instance = instance;
+
+        this._initPositioningStrategy();
+    },
+
+    _initPositioningStrategy: function() {
```

Comment:
May be 
this._positioningStrategy = this.instance.fire("isAdaptive") ? new AdaptivePositioningStrategy(this) : new BasePositioningStrategy(this);